src/Events/Notifications/Provider.php

```diff
@@ -25,20 +25,8 @@ class Provider extends Service_Provider {
 	 * @since 6.4.0
 	 */
 	public function register() {
-		add_action( 'admin_footer', [ $this, 'render_icon' ] );
-	}
-
-	/**
-	 * Renders the Notification icon.
-	 *
-	 * @since 6.4.0
-	 */
-	public function render_icon() {
-		// TODO: Remove this check when we have a proper way to register it, like register_on_action.
-		if ( ! class_exists( Notifications::class ) ) {
-			return;
-		}
+		$this->container->singleton( static::class, $this );
```

Comment:
Architecture choices of a provider binding itself i think are confusing. Can we bind this provider 1 (or more) level up ?